ccw.core/src/java/ccw/CCWPlugin.java

```diff
@@ -109,31 +105,18 @@ public static ITracer getTracer() {
 			return NullTracer.INSTANCE;
 	}
     
-    public synchronized void startREPLServer() {
-    	if (ackREPLServer == null) {
-	        try {
-	        	// TODO use ClojureOSGi.withBundle instead
-	        	Var startServer = BundleUtils.requireAndGetVar(getBundle().getSymbolicName(), "clojure.tools.nrepl.server/start-server");
-	        	Object defaultHandler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.server/default-handler").invoke();
-	        	Object handler = BundleUtils.requireAndGetVar(
-	        	        getBundle().getSymbolicName(),
-	        	        "clojure.tools.nrepl.ack/handle-ack").invoke(defaultHandler);
-	            ackREPLServer = (ServerSocket)((Map)startServer.invoke(Keyword.intern("handler"), handler)).get(Keyword.intern("server-socket"));
-	            CCWPlugin.log("Started ccw nREPL server: nrepl://127.0.0.1:" + ackREPLServer.getLocalPort());
-	        } catch (Exception e) {
-	            CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
-	            throw new RuntimeException("Could not start plugin-hosted REPL server", e);
-	        }
+    public void startREPLServer() {
+    	try {
+    	    ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-start-if-necessary");
+    	} catch (Exception e) {
+    		CCWPlugin.logError("Could not start plugin-hosted REPL server", e);
+    		throw new RuntimeException("Could not start plugin-hosted REPL server", e);
     	}
     }
     
     public int getREPLServerPort() {
-    	if (ackREPLServer == null) {
-    		startREPLServer();
-    	}
-    	return ackREPLServer.getLocalPort();
+    	startREPLServer();
+    	return (Integer) ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-port");
```

Comment:
Ok, so now please don't touch / rebase your branch, will be merged very
sooooon

2015-03-25 10:38 GMT+01:00 Andrea Richiardi <notifications@github.com>:

> In ccw.core/src/java/ccw/CCWPlugin.java
> <https://github.com/laurentpetit/ccw/pull/36#discussion_r27105506>:
>
> >      	}
> >      }
> >
> >      public int getREPLServerPort() {
> > -    	if (ackREPLServer == null) {
> > -    		startREPLServer();
> > -    	}
> > -    	return ackREPLServer.getLocalPort();
> > +    	startREPLServer();
> > +    	return (Integer) ClojureInvoker.newInvoker(this, "ccw.core.launch")._("ccw-nrepl-port");
>
> None, done ;) I would like to try now the cider branch, because probably
> the problem I had was because of the bug in launch.clj
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/laurentpetit/ccw/pull/36/files#r27105506>.
>



-- 
Laurent Petit